src/app/hook/useObterUUIDEmpresaDaURL.ts

```diff
@@ -0,0 +1,9 @@
+import { useHistory } from 'react-router-dom';
+
+export default function useObterUUIDEmpresaDaURL(): string | undefined {
+  const { location: { pathname }} = useHistory();
+
+  return pathname
+    .match(/[a-f0-9]{8}-[a-f0-9]{4}-4[a-f0-9]{3}-[89aAbB][a-f0-9]{3}-[a-f0-9]{12}/)
+    ?.shift();
```

Comment:
Acha que vai utilizar essa string de match de uuid em algum outro lugar? Sepa era bom deixar uma constante com nome descritivo exportada em algum lugar se for.

Reply:
po, se for usar pra pegar o UUID pode usar esse hook, se colocar o regex em uma constante pode remover o hook.

Qual eh melhor, constant ou hook?

Reply:
Eu estava falando em outros casos, porque esse hook vai extrair somente da url. Como não tem outro lugar usando deixa pra extrair quando precisar.